@here/harp-geoutils/lib/projection/Projection.ts

```diff
@@ -123,43 +123,93 @@ export interface Projection {
      */
-    surfaceNormal(worldPoint: Vector3Like): Vector3Like;
+    abstract surfaceNormal(worldPoint: Vector3Like): Vector3Like;
 
     /**
      * Returns the surface normal at the given world position.
      *
      * @param worldPoint The position in world coordinates.
      * @returns The resulting normal vector.
      */
-    surfaceNormal<Normal extends Vector3Like>(worldPoint: Vector3Like, result: Normal): Normal;
+    abstract surfaceNormal<Normal extends Vector3Like>(
+        worldPoint: Vector3Like,
+        result: Normal
+    ): Normal;
 
     /**
      * Returns the signed distance between the given coordinates and
      * the closest point on the surface.
      *
      * @param worldPoint The position in world coordinates.
      */
-    groundDistance(worldPoint: Vector3Like): number;
+    abstract groundDistance(worldPoint: Vector3Like): number;
 
     /**
      * Scales the given world coordinates to the surface.
      *
      * @param worldPoint The position in world coordinates.
      */
-    scalePointToSurface(worldPoint: Vector3Like): Vector3Like;
+    abstract scalePointToSurface(worldPoint: Vector3Like): Vector3Like;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid world position for the given source projection.
+     * @returns The world position reprojected using this [[Projection]].
+     */
+    reprojectPoint(sourceProjection: Projection, worldPos: Vector3Like): Vector3Like;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid position in the world space defined by the source projection.
+     * @param result The resulting position reprojected using this [[Projection]].
+     */
+    reprojectPoint<WorldCoordinates extends Vector3Like>(
+        sourceProjection: Projection,
+        worldPos: Vector3Like,
+        result: WorldCoordinates
+    ): typeof result;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid position in the world space defined by the source projection.
+     * @param result The resulting position reprojected using this [[Projection]].
+     * @hidden
+     */
+    reprojectPoint(
+        sourceProjection: Projection,
+        worldPos: Vector3Like,
+        result?: Vector3Like
+    ): Vector3Like {
+        if (sourceProjection === this) {
+            if (result === undefined) {
+                return { x: worldPos.x, y: worldPos.y, z: worldPos.z };
+            }
+            result.x = worldPos.x;
+            result.y = worldPos.y;
+            result.z = worldPos.z;
+            return result;
+        }
+        return this.projectPoint(sourceProjection.unprojectPoint(worldPos), result);
```

Comment:
you mentioned that there are some short cuts to transform between different projections, i.e. webmercator -> sphere, how do you handle that with this API?

Reply:
I was thinking to do something like this:
```
class SphereProjection extends Projection {
    // OK, reprojectPoint is compatible with the runtime method signature of Projection.
    // and since SphereProjection is not exported with don't have to care about wrong
    // usages of Generics.
    reprojectPoint(sourceProjection: Projection,
        worldPos: Vector3Like, result?: Vector3Like): Vector3Like {
        if (sourceProjection === mercatorProjection) {
            ...
        }
        ...
    }
```

Reply:
If needed we could also maintain a global list of `converters` so users could inject their own reprojection and we could call it in the default implementation of reprojectPoint. I'm not sure it is need, but the following should work:
```
export abstract class Projection {
   ...
    static registerReprojection(...): void {
        reprojections.push(...)
    }

    // ... reprojectPoint restricted signatures...

    // runtime reprojectPoint method
    reprojectPoint(sourceProjection: Projection, ...) {
        ...
        const reprojection = Projection.findReprojection(this, sourceProjection);
        if (reprojection) ... else ...
    }
```